app/controllers/errors_controller.rb

```diff
@@ -2,7 +2,10 @@
 
 class ErrorsController < ApplicationController
   def not_found
-    render status: 404
+    respond_to do |format|
+      format.html { render status: 404 }
+      format.all { render status: 404, formats: :html, content_type: 'text/html' }
```

Comment:
```suggestion
      format.any { render status: 404, formats: :html, content_type: 'text/html' }
```

"Respond to also allows you to specify a common block for different formats by using any:" [respond_to](https://apidock.com/rails/ActionController/MimeResponds/respond_to)

this seems to work, did you try it before and it didn't?